Fixes for tasks 211 and 215

hvaara opened the pull request:
This PR adds `200 OK` status code for the mocks in task 211 and 215. For 215 it also adds the option to use `response.json()` and fixes a spelling error in the description.

**Test results for `starcoder2-15b`:**

- **BigCodeBench/211:** Progressed past the HTTP error, but saves the zip file to the current directory as opposed to the destination directory. It is actually not specified in the description of the task where to save the zip file. This might be addressed with and update to the task specification or the tests. I'll leave it as future work. Another, bigger, issue is that the zip file is removed with `os.remove('temp.zip')` at the end, which is against the task description, so the model should fail this task based on my understanding.
  
- **BigCodeBench/215:** Issue resolved through updates to the tests. `starcoder2-15b` now passes this test in BigCodeBench.

Fixes #33

terryyz commented:
@hvaara Let me know how it goes with the model verification :) If it works, please put it into `tools/fix_0.2.1` and point to the data to `v0.1.3`.

hvaara commented:
@terryyz I'm working on testing resolution with starcoder2-15b where the bug was detected. Which fix file, `BIGCODEBENCH_VERSION` and `BIGCODEBENCH_NEW_VERSION` should I be targeting?

terryyz commented:
Thanks @hvaara! Could you move the part to a file called `fix_v023.py`, which will be a patch based on the  v0.2.3 release.

hvaara commented:
@terryyz Thanks for the review! I moved the changes to a new file and updated the minor versions of `BIGCODEBENCH_VERSION` and `BIGCODEBENCH_NEW_VERSION`.
